crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -798,17 +799,72 @@ mod tests {
     const MAX_PARKED_TXS_PER_ACCOUNT: usize = 15;
     const TX_TTL: Duration = Duration::from_secs(2);
 
-    fn mock_ttx(
+    struct MockTTXBuilder {
         nonce: u32,
-        signer: &SigningKey,
+        signer: SigningKey,
         denom_0_cost: u128,
         denom_1_cost: u128,
         denom_2_cost: u128,
-    ) -> TimemarkedTransaction {
-        TimemarkedTransaction::new(
-            mock_tx(nonce, signer, "test"),
-            mock_tx_cost(denom_0_cost, denom_1_cost, denom_2_cost),
-        )
+    }
+
+    impl MockTTXBuilder {
+        fn build(self) -> TimemarkedTransaction {
+            let tx = mock_tx(self.nonce, &self.signer, "test");
+
+            TimemarkedTransaction::new(
+                tx,
+                mock_tx_cost(self.denom_0_cost, self.denom_1_cost, self.denom_2_cost),
+            )
+        }
+
+        fn new() -> Self {
+            Self::default()
+        }
+
+        fn default() -> Self {
+            Self {
+                nonce: 0,
+                signer: get_alice_signing_key(),
+                denom_0_cost: 0,
+                denom_1_cost: 0,
+                denom_2_cost: 0,
+            }
+        }
+
+        fn nonce(self, nonce: u32) -> Self {
+            Self {
+                nonce,
+                ..self
+            }
+        }
+
+        fn signer(self, signer: SigningKey) -> Self {
+            Self {
+                signer,
+                ..self
+            }
+        }
+
+        fn denom_0(self, denom_0_cost: u128) -> Self {
```

Comment:
I had a hard time grokking these. IMO stay closer to the original `cost` of the time marked transaction. I would be able to immediately grep the code base.

You can add these for more elegance:
```rust
fn cost_map(self, cost: HashMap<IbcPrefixed, u128>) -> Self {
    Self {
        cost: cost,
        ..self
    }
}

fn set_cost(self, key: IbcPrefixed, value: u128) -> Self {
    self.cost[key] = value;
    self
}
```
We use this pattern throughout the codebase, so you could also do that:
```rust
fn set_cost<T: Into<IbcPrefixed>>(self, key: T, value: u128) -> Self {
    self.cost[key.into()] = value;
    self
}
```